src/test/java/org/openelisglobal/action/action.java

```diff
@@ -0,0 +1,130 @@
+package org.openelisglobal.action;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+
+import org.junit.jupiter.api.*;
+import org.openelisglobal.action.service.ActionService;
+import org.openelisglobal.action.valueholder.Action;
+import org.openelisglobal.BaseWebContextSensitiveTest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.Statement;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Transactional
+public class ActionServiceTest extends BaseWebContextSensitiveTest {
+
+    @Autowired
+    private ActionService actionService;
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Autowired
+    private DataSource dataSource;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        executeDataSetWithStateManagement("testdata/action.xml");
+    }
+
+    @AfterEach
+    public void cleanUp() throws Exception {
+        try (Connection conn = dataSource.getConnection(); Statement stmt = conn.createStatement()) {
+            stmt.execute("TRUNCATE TABLE action RESTART IDENTITY CASCADE");
```

Comment:
can you use the existing [leanRowsInCurrentConnection util](https://github.com/DIGI-UW/OpenELIS-Global-2/blob/develop/src/test/java/org/openelisglobal/BaseWebContextSensitiveTest.java#L119) to clean the tables ??